src/components/biggive-tipping-slider/biggive-tipping-slider.tsx

```diff
@@ -0,0 +1,117 @@
+import { Component, Element, Prop, h } from '@stencil/core';
+import { spacingOption } from '../../globals/spacing-option';
+import { brandColour } from '../../globals/brand-colour';
+
+@Component({
+  tag: 'biggive-tipping-slider',
+  styleUrl: 'biggive-tipping-slider.scss',
+  shadow: true,
+})
+export class BiggiveTippingSlider {
+  @Element() host: HTMLBiggiveTippingSliderElement;
+
+  @Prop() spaceBelow: spacingOption = 0;
+
+  @Prop() colourScheme: brandColour = 'primary';
+
+  @Prop() percentageStart: number;
+
+  @Prop() percentageEnd: number;
+
+  @Prop() donationAmount: number;
+
+  @Prop() donationCurrency: string = '£';
+
+  componentDidRender() {
+    var isMoving = false;
+    var handle = this.host.shadowRoot?.querySelector<HTMLElement>('.handle');
+    var bar = this.host.shadowRoot?.querySelector<HTMLElement>('.bar');
+    var percentageWrap = handle?.querySelector('.percentage-value');
+    var donationWrap = handle?.querySelector('.donation-value');
+
+    var move = e => {
+      if (isMoving && bar && handle && percentageWrap && donationWrap) {
+        var min = 0;
+        var max = bar.offsetWidth - handle.offsetWidth;
+        var pageX = e.type == 'touchstart' || e.type == 'touchmove' ? e.touches[0].pageX : e.pageX;
+        var mousePos = pageX - bar?.offsetLeft - handle?.offsetWidth / 2;
+        var position = mousePos > max ? max : mousePos < min ? min : mousePos;
+        var percentage = (position / max) * this.percentageEnd;
+        var donation = Math.round(this.donationAmount * (percentage / 100));
+
+        percentageWrap.innerHTML = Math.round(percentage).toString();
+        donationWrap.innerHTML = Math.round(donation).toString();
+
+        handle.style.marginLeft = position + 'px';
+      }
+    };
+
+    bar?.addEventListener('mousedown', function (e) {
```

Comment:
Shouldn't we crash if we don't have a `bar`, instead of just continuing like everything's fine? Maybe we want to assert its not null immediately when we call `querySelector` to fetch it.

Reply:
We know the bar is there - so adding the non-assertion operator to the definition allows us to remove the ?

Reply:
Yep, should do.